src/components/Calendar/Filters.tsx

```diff
@@ -0,0 +1,70 @@
+"use client";
+
+import { Checkbox } from "@mui/material";
+import React from "react";
+import styled from "styled-components";
+
+interface CalendarFilterAccordionProps {
+    allLocations: string[];
+    editLocations: (locations: string[]) => void;
+    currentLocations: string[];
+}
+
+const ContainerDiv = styled.div`
+    display: grid;
+    margin-bottom: 1rem;
+    justify-content: center;
+    background-color: ${(props) => props.theme.main.background[2]};
+    color: ${(props) => props.theme.main.foreground[2]};
+    padding: 1rem;
+    border-radius: 1rem;
+    border: 1px solid ${(props) => props.theme.main.border};
+    grid-template-columns: 1fr;
+    @media (min-width: 700px) {
+        grid-template-columns: repeat(2, 1fr);
+    @media (min-width: 1100px) {
+        grid-template-columns: repeat(3, 1fr);
+
+`;
+
+const CheckboxAndTitleDiv = styled.div`
+    display: flex;
+    gap: 0.5rem;
+    align-items: center;
+    min-width: 300px;
+`;
+
+const CalendarFilters: React.FC<CalendarFilterAccordionProps> = ({
+    allLocations,
+    currentLocations,
+    editLocations,
+}) => {
+    return (
+        <ContainerDiv>
+            <h2>Shown Locations:</h2>
+            {allLocations.map((location) => {
+                return (
+                    <CheckboxAndTitleDiv key={location}>
+                        <Checkbox
+                            checked={currentLocations.includes(location)}
+                            onChange={(event) => {
```

Comment:
This is quite a big function. Maybe it is better to have it outside of the return statement or even the component itself.

Reply:
I personally think it's ok? Idk seeing as the rest of it isn't long

Reply:
Idk tho I'll see what others think I'm obvs happy to change it!